app/controllers/dams_resource_controller.rb

```diff
@@ -188,7 +190,22 @@ def solr_escape (str)
     pattern = /(\+|\-|\&\&|\|\||\!|\(\)|\{\}|\[|\]|\^|\"|\~|\*|\?|\:|\\)/
     str.gsub(pattern){|match|"\\"  + match} 
   end
-
+  
+  def replace_config_file (file_name)
+    copy_config_file (file_name)
+    config = ERB.new(IO.read("#{APP_ROOT}/config/#{file_name}")).result(binding)
+    File.open("#{APP_ROOT}/config/#{file_name}", "w") do |f|
+      f.write(config)
+    end
+  end
+  
+  def copy_config_file (file_name)
+    data = File.read("#{APP_ROOT}/config/#{Rails.configuration.share_notify_sample}")
+    File.open("#{APP_ROOT}/config/#{file_name}", "w") do |fi|
+      fi.write(data)
+    end
```

Comment:
@hweng yes, I need it.  Here's the logic of this code, the first time it calls replace config file function. 

https://github.com/ucsdlib/damspas/blob/feature/new_deploy/app/controllers/dams_resource_controller.rb#L230
It will use the rails.config env to copy the yml file with correct env variables - either share_notify.yml.prod.sample or share_notify.yml.staging.sample  into the share_notify.yml.  The share_notify.yml will contain for example this "ENV.fetch('APPS_DHH_STAGING_SHARE_NOTIFY_HOST')". Then it will load the correct environment values, and it will write that data into itself share_notify.yml.  Now if you view share_notify.yml , it will contain the correct value for example https://staging.osf.io and token value instead.  After share_notify gem used it, if we don't copy back the sample one, then that become different file, git status will tell us that it is modified and if we commit that in, then that file no longer contains the env variable to be replaced and it contain our secret env variable values. For production, it's ok to not call this line, since we don't commit change from prod. However, for other environments we do need that.
https://github.com/ucsdlib/damspas/blob/feature/new_deploy/app/controllers/dams_resource_controller.rb#L233